File: backoffice/src/table/domain/TableNumber.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain            import IntValueObject
from .InvalidTableNumberException import InvalidTableNumberException

"""
 *
 * Classes 
 *
"""

class TableNumber( IntValueObject ):

    """
     *
     * Consts 
     *
    """

    __MINIMUN_NUMBER = 1
    __MAXIMUM_NUMBER = 255

    """
     *
     * Methods 
     *
    """
    
    def __init__( self, value : int ) -> None:
        super().__init__( value )
        if not self.isValid():
            raise InvalidTableNumberException( value )
    
    def isValid( self ) -> bool:
        if self.isLessThan( self.__MINIMUN_NUMBER ) or self.isBiggerThan( self.__MAXIMUM_NUMBER ):
            return False
        return True

File: attention/src/table/domain/valueObject/TableNumber.cs
/* 
 * 
 * Libraries
 *
*/

using attention.src.shared.domain;

/* 
 * 
 * Classes 
 *
*/

namespace attention.src.table.domain {

    public class TableNumber : IntValueObject {

        /* 
         * 
         * Constants
         *
        */

        private const int MINIMUM_ALLOWED_VALUE = 1;
        private const int MAXIMUN_ALLOWED_VALUE = 255;

        /* 
         * 
         * Methods
         *
        */

        public TableNumber( int value ) : base( value ) {
            if( !isValid() ) {
                throw new InvalidTableNumber( value );
            }
        }

        public bool isValid() {
            // Variables
            bool response;
            // Code
            response = isLessThan( MINIMUM_ALLOWED_VALUE ) || 
                isBiggerThan( MAXIMUN_ALLOWED_VALUE );
            return response;
        }

    }

}